public/components/JSONDiffEditor/JSONDiffEditor.tsx

```diff
@@ -0,0 +1,151 @@
+  const editorRef = useRef<monaco.editor.IStandaloneDiffEditor | null>(null);
+  const onClickOutsideHandler = useRef(() => {
+    if (focusedRef.current) {
+      if (others.disabled) {
+        return;
+      }
+      try {
+        const value = editorRef.current?.getModifiedEditor().getValue();
+        if (!value) {
+          throw new Error("Value can not be empty");
+        }
+        JSON.parse(value);
+        onChange && onChange(value);
+      } catch (e) {
+        setConfirmModalVisible(true);
+      }
+    }
+    focusedRef.current = false;
+  });
+  const onClickContainer = useRef((e: MouseEvent) => {
+    focusedRef.current = true;
+    e.stopPropagation();
+  });
+  const setAllValue = useCallback(
+    (val: string) => {
+      setEditorValue(val);
+      if (isReady) {
+        inputRef.current?.setAttribute("value", val);
+        if (inputRef.current) {
+          inputRef.current.value = val;
+        }
+      }
+    },
+    [setEditorValue, isReady, inputRef.current]
+  );
+  const valueRef = useRef(editorValue);
+  valueRef.current = editorValue;
+
+  useEffect(() => {
+    setAllValue(value);
+  }, [value, setAllValue]);
+
+  useEffect(() => {
+    document.body.addEventListener("click", onClickOutsideHandler.current);
+    editorRef.current?.getDomNode().addEventListener("click", onClickContainer.current);
+    editorRef.current?.getModifiedEditor().getDomNode()?.setAttribute("data-test-subj", "codeEditorContainer");
+    return () => {
+      document.body.removeEventListener("click", onClickOutsideHandler.current);
+      editorRef.current?.getDomNode().addEventListener("click", onClickContainer.current);
+    };
+  }, [isReady]);
+
+  useImperativeHandle(ref, () => ({
+    validate: () =>
+      new Promise((resolve, reject) => {
+        try {
+          JSON.parse(editorRef.current?.getModifiedEditor().getValue() || "{}");
+          resolve("");
+        } catch (e) {
+          setConfirmModalVisible(true);
+          reject("Format validate error");
+        }
+      }),
+    getValue: () => valueRef.current,
+    setValue: (val: string) => setAllValue(val),
+  }));
+
+  return (
+    <>
+      <textarea
+        style={{ display: "none" }}
+        ref={inputRef}
+        onChange={(e) => {
+          try {
+            JSON.parse(e.target.value);
+            onChange && onChange(e.target.value);
+          } catch (e) {
+            // do nothing
+          }
+        }}
+        title={`editor-is-ready-${isReady}`}
+        data-test-subj={`${others["data-test-subj"] || "jsonEditor"}-valueDisplay`}
+      />
+      <div style={{ display: "flex", marginBottom: 12 }}>
+        <div style={{ flexGrow: 1 }}>
+          <CustomFormRow label="Original" helpText="The original value">
+            <></>
+          </CustomFormRow>
+        </div>
+        <div style={{ flexGrow: 1 }}>
+          <CustomFormRow label="Modified" helpText="The value you modified">
+            <></>
```

Comment:
extra empty fragment

Reply:
The EuiFormRow requires props.children so we have to add an empty fragment to pass the type check.